tools/osbuilder/image-builder/image_builder.sh

```diff
@@ -186,7 +186,7 @@ check_rootfs() {
 	OK "init is installed"
 
 
-	systemd_path="/lib/systemd/systemd"
+	systemd_path="/usr/lib/systemd/systemd"
```

Comment:
ehmmm no, this won't work in other distributions, like ubuntu or debian

```
|X@ubuntu ~ 
└─ (ubuntu) $ ls -l /usr/lib/systemd/systemd
ls: cannot access '/usr/lib/systemd/systemd': No such file or directory
```

Reply:
@devimc, let me guess, you tested using a container, didn't you?

On a Ubuntu 20.04 VM:
```
ubuntu@ubuntu:~$ ls -lha /usr/lib/systemd/systemd
-rwxr-xr-x 1 root root 1.6M Jul  6 21:38 /usr/lib/systemd/systemd
```

Reply:
tested on bare metal, ubuntu 18.04

Reply:
And does `/lib/systemd/systemd` exist?

Reply:
it does, 

```
$ ls -l /lib/systemd/systemd
-rwxr-xr-x 1 root root 1616248 Jul  8 18:59 /lib/systemd/systemd
```

Reply:
@pmores, can you spawn some VMs and check this out?

Reply:
FWIW, no `/usr/lib/systemd` on `ubuntu:latest` in a container:

```
podman run -it ubuntu:latest bash
root@12552a96f810:/# ls -l /lib
total 24
drwxr-xr-x.  2 root root 4096 May 15  2019 init
drwxr-xr-x.  3 root root 4096 May 15  2019 lsb
drwxr-xr-x.  3 root root 4096 Mar 15  2018 systemd
drwxr-xr-x. 15 root root 4096 Feb 12  2018 terminfo
drwxr-xr-x.  2 root root 4096 May 15  2019 udev
drwxr-xr-x.  3 root root 4096 May 15  2019 x86_64-linux-gnu
root@12552a96f810:/# ls -l /lib/systemd/
total 4
drwxr-xr-x. 2 root root 4096 May 15  2019 system
root@12552a96f810:/# ls -l /usr/lib/systemd
ls: cannot access '/usr/lib/systemd': No such file or directory
root@12552a96f810:/# 
```

Reply:
I based my assumptions on Debian 10 and Ubuntu 19.04 both of which have /lib symlinked to /usr/lib.  Because of this, both paths (/lib/systemd/systemd and /usr/lib/systemd/systemd) resolve on those systems with /usr/lib/systemd/systemd being the canonical one.

I guess the real question here, is there a rule or a standard governing this to rely on?  If not I'm afraid it might be necessary just to check both (or all, if any other emerge in the future?) candidate paths.  Is there a simpler solution?

Reply:
@pmores, seems tha checking for both paths (which makes things slightly more complicated) may be the way to go, I don't see a simpler solution than that, unfortunately.

Reply:
I think we should check for both paths

Reply:
@fidencio @devimc Thanks for sharing your opinion guys, I'll submit a PR to generalise this shortly!

Reply:
thanks @pmores and apologies for not reviewing this sooner (version 1.x)